app/segments/drill/[id]/submission/input.js

```diff
@@ -137,6 +141,136 @@ async function uploadAttempt(
+function checkLeaderboardUpdate(uploadData, drillInfo, currentLeaderboard) {
+  const leaderboardData = currentLeaderboard["data"];
+
+  //used if an attempt already exists
+  const currentBest =
+    leaderboardData[uploadData.uid]?.[drillInfo.mainOutputAttempt]?.value;
+
+  const newAttempt = {
+    [drillInfo.mainOutputAttempt]: {
+      id: uploadData.id,
+      value: uploadData[drillInfo.mainOutputAttempt],
+    },
+  };
+
+  console.log("currentBest: ", currentBest);
+
+  //check if the user exists on the leaderboard
+  if (leaderboardData[uploadData.uid] == undefined) {
+    console.log("User not on leaderboard, uploading this attempt");
+
+    uploadNewLeaderboard(
+      leaderboardData,
+      newAttempt,
+      uploadData.uid,
+      uploadData.did,
+    );
+  } else {
+    //Determine if lower is better for mainOutputAttempt
+    if (drillInfo.aggOutputs[drillInfo.mainOutputAttempt].lowerIsBetter) {
+      console.log("Lower is in fact better");
+      //compare the current attempt to the attempt currently on leaderboard
+      if (uploadData[drillInfo.mainOutputAttempt] < currentBest) {
+        console.log("New Best Attempt! Time to upload!");
+
+        uploadNewLeaderboard(
+          leaderboardData,
+          newAttempt,
+          uploadData.uid,
+          uploadData.did,
+        );
+      }
+      //else for testing
+      else {
+        console.log("Didn't update");
+      }
+    } else {
+      // console.log(uploadData)
+      console.log("Lower is not better");
+      //compare the current attempt to the attempt currently on leaderboard
+      if (uploadData[drillInfo.mainOutputAttempt] > currentBest) {
+        console.log("New Best Attempt! Time to upload!");
+
+        uploadNewLeaderboard(
+          leaderboardData,
+          newAttempt,
+          uploadData.uid,
+          uploadData.did,
+        );
+      }
+      //else for testing
+      else {
+        console.log("Didn't update");
+      }
+    }
+  }
+}
+
+async function uploadNewLeaderboard(leaderboardData, newAttempt, uid, did) {
+  //Add new attempt to leaderboard
+  leaderboardData[uid] = newAttempt;
+  const submitMainOutputAttempt = Object.keys(newAttempt)[0];
+  console.log("new attempt");
+  console.log(newAttempt);
+
+  //Reference to best_attempts drill document
+  const bestAttemptsDrillRef = doc(db, "teams", "1", "best_attempts", did);
+
+  try {
+    console.log("LEADERBOARD UPDATE STARTED");
+    console.log(submitMainOutputAttempt);
+
+    // get a fresh call without tanstack cache just in case
+    const latestLeaderboard = await getDoc(bestAttemptsDrillRef);
+    const latestLeaderboardData = latestLeaderboard.data() || {};
+    console.log("maybe cached leaderboard");
+    console.log(leaderboardData);
+    console.log("latest leaderboard");
+    console.log(latestLeaderboardData);
+    console.log("number of leaderboard entries");
+    console.log(Object.keys(latestLeaderboardData).length);
+
+    if (latestLeaderboard.exists()) {
```

Comment:
Merge these 2 if cases because the else case is doing the same thing

Reply:
In general, do you think it's necessary to handle the possibility of multiple mainOutputAttempt values (e.g. sorting by proxHoleAverage or StrokesGainedAverage for the same leaderboard) at the moment?

I was trying to address that edge case but I think I may have mixed up some implementation details (based on how `newAttempt` is assigned).

Reply:
I don't think mainOutputAttempt is addressed here. I think it's fine to ignore mainOutputAttempt because the biggest hurdle to switching from a single mainOutputAttempt to multiple, namely the database, is already set up for that purpose. Think of it like the teams switching feature

Reply:
For future reference, I decided to defer these 2 edge cases to future PR (out of scope atm):
- multiple mainOutputAttempt values per drill ID
- best_attempts > drillId is undefined (no automation for this yet)